serveradmin/api/decorators.py

```diff
@@ -55,15 +55,22 @@ def _wrapper(request):
         then = datetime.utcfromtimestamp(
             int(request.META['HTTP_X_TIMESTAMP'])
         ).replace(tzinfo=timezone.utc)
-
-        app = authenticate_app(
-            public_keys, signatures, app_id, token, then, now, body
-        )
-
         body_json = json.loads(body) if body else None
+        status_code = 200
+
         try:
-            status_code = 200
+            app = authenticate_app(
+                public_keys, signatures, app_id, token, then, now, body
+            )
```

Comment:
So here's what this will do:
1) make sure we return a proper JSON API response on auth errors instead of the default django error page
2) tell the client _why_ the authentication was forbidden.

I'm not quite sure about the security implications of 2, though I think it should be fine. 1 is definitely a good thing!

Reply:
Sorry I am confused. Do you consider this now correct or no ?

Reply:
I think it's probably a good idea, as discussed in the call.

Reply:
This turns out to be just a summary/explanation of the desired changes.